plugin/pkg/shoot/validator/admission.go

```diff
@@ -355,9 +355,13 @@ func (c *validationContext) validateProjectMembership(a admission.Attributes) er
 }
 
 // validateSeedSelectionForHAShoot validates the seed assigned for the shoot based on the following criteria:
-// 1. A HA shoot with failure tolerance of 'zone' can only be scheduled on multi-zonal seeds
-// 2. A HA shoot with failure tolerance of 'node' or a non-HA shoot can be scheduled onto a multi-zonal as well as non-multi-zonal seed.
+// 1. A HA shoot can be scheduled on a seed with HA enabled only
```

Comment:
As far as I understand it `HAControlPlanes` feature in gardenlet controls the number of replicas of the shoots control plane components rather than the HA configuration of the seeds itself.
I'm able to bootstrap a gardenlet with HA control plane "true on a cluster without and found this in botanist.

https://github.com/gardener/gardener/blob/8f48339c5b5be7ca96072dbe680b9e7935ebe811/pkg/operation/botanist/etcd.go#L268-L273
https://github.com/gardener/gardener/blob/8f48339c5b5be7ca96072dbe680b9e7935ebe811/pkg/operation/botanist/kubeapiserver.go#L303-L305

In all my tests where the gardenlet HA control plane feature was "false", shoots with failure tolerance "node" node have been created with one replica of etcd and apiserver.

Reply:
Ok in other words:

- for a shoot with controlPlane.failureTolerance.type = zone => a minimum requirement is that the seed should have spec.highAvailability.type = zone
- for a shoot with controlPlane.failureTolerance.type = node => a minimum requirement is that the seed should have spec.highAvailability.type = node. in other words the seed could be set up with failureTolerance.type = node | zone.